src/sentry/snuba/discover.py

```diff
@@ -945,3 +946,143 @@ def get_facets(query, params, limit=10, referrer=None):
+    )
+    histogram_col = get_measurements_histogram_col(histogram_params)
+
+    results = query(
+        selected_columns=[key_col, histogram_col, "count()"],
+        conditions=[[get_function_alias(key_col), "IN", measurements]],
+        query=user_query,
+        params=params,
+        # the zerofill step assumes it is ordered by the bin then name
+        orderby=[get_function_alias(histogram_col), get_function_alias(key_col)],
+        limit=len(measurements) * num_buckets,
+        referrer=referrer,
+        auto_fields=True,
+        use_aggregate_conditions=True,
+    )
+
+    return normalize_measurements_histogram(
+        measurements, num_buckets, key_col, histogram_params, results
+    )
+
+
+def get_measurements_histogram_col(params):
+    return u"measurements_histogram({:d}, {:d}, {:d})".format(*params)
+
+
+def find_measurements_histogram_params(
+    measurements, num_buckets, min_value, max_value, precision, query, params
+):
+    multiplier = int(10 ** precision)
+
+    # finding the bounds might result in None if there isn't sufficient data
+    if min_value is None or max_value is None:
+        return HistogramParams(1, 0, multiplier)
+
+    # A single bucket in a histogram has contains values from [start, end),
+    # meaning that start is inclusive and end is exclusive
+
+    scaled_min = int(floor(multiplier * min_value))
+    scaled_max = int(ceil(multiplier * max_value))
+
+    # align the first bin with the minimum value
+    start_offset = scaled_min
+
+    bucket_size = int(ceil((scaled_max - scaled_min) / float(num_buckets)))
+    if bucket_size == 0:
+        bucket_size = 1
+    # Sometimes the max value lies on the bucket boundary, and since the end
+    # of the bucket is exclusive, it gets excluded. To account for that, we
+    # increase the width of the buckets to cover the max value.
+    if start_offset + num_buckets * bucket_size <= max_value:
+        bucket_size += 1
+
+    return HistogramParams(bucket_size, start_offset, multiplier)
+
+
+def find_measurements_min_max(measurements, user_query, params):
+    min_columns, max_columns = [], []
+    for measurement in measurements:
+        min_columns.append("min(measurements.{})".format(measurement))
+        max_columns.append("max(measurements.{})".format(measurement))
+
+    results = query(
+        selected_columns=min_columns + max_columns,
+        query=user_query,
+        params=params,
+        limit=1,
+        referrer="api.organization-events-measurements-min-max",
+        auto_fields=True,
+        use_aggregate_conditions=True,
+    )
+    # there should be exactly 1 row in the results
+    data = results["data"][0]
+
+    min_value = None
+    for min_column in min_columns:
+        value = data[get_function_alias(min_column)]
+        if value is None:
+            continue
+        if min_value is None or value < min_value:
+            min_value = value
```

Comment:
If you put all the min values into a list you could use `min(values)` to get the lowest min. It might shave a few lines of code off. Something like

```python
values = [data[get_function_alias(column)] for column in min_columns]
min_value = min(filter(lambda v: v is not None, values))
```

Reply:
Addressed in 4646cf4